internal/venice-common/src/main/java/com/linkedin/venice/serialization/avro/AvroSpecificStoreDeserializerCache.java

```diff
@@ -27,6 +27,10 @@ public AvroSpecificStoreDeserializerCache(
     this(id -> schemaRepository.getValueSchema(storeName, id).getSchema(), valueClass);
   }
 
+  public AvroSpecificStoreDeserializerCache(Schema valueSchema, Class<V> valueClass) {
```

Comment:
This... is kind of sketchy.  It's sketchy enough that I think we should extend this class to include it perhaps.  This is creating an instance of the deserializer cache that will only return one version of the value schema regardless of what id you pass to it.  I need to wrap my head around this PR more to understand what this is for I think.